src/NuGet.Clients/PackageManagement.VisualStudio/Migrate/ProjectJsonToPackageRefMigrator.cs

```diff
@@ -0,0 +1,146 @@
+                        CultureInfo.CurrentCulture,
+                        Strings.Error_MultipleFrameworksUwp,
+                        _project.MSBuildProjectPath));
+            }
+
+            var dependencies = new List<LibraryDependency>();
+            foreach (var targetFramework in packageSpec.TargetFrameworks)
+            {
+                if (targetFramework.FrameworkName != NuGetFramework.Parse("uap10.0"))
+                {
+                    throw new InvalidOperationException(
+                    string.Format(
+                        CultureInfo.CurrentCulture,
+                        Strings.Error_TargetFrameworkNotUwp,
+                        _project.MSBuildProjectPath));
+                }
+                dependencies.AddRange(targetFramework.Dependencies);
+
+            }
+
+            dependencies.AddRange(packageSpec.Dependencies);
+            foreach (var dependency in dependencies)
+            {
+                var includeFlags = dependency.IncludeType;
+                var privateAssetsFlag = dependency.SuppressParent;
+                var metadataElements = new List<string>();
+                var metadataValues = new List<string>();
+                if (includeFlags != LibraryIncludeFlags.All)
+                {
+                    metadataElements.Add("IncludeAssets");
+                    metadataValues.Add(LibraryIncludeFlagUtils.GetFlagString(includeFlags));
+                }
+
+                if (privateAssetsFlag != LibraryIncludeFlagUtils.DefaultSuppressParent)
+                {
+                    metadataElements.Add("PrivateAssets");
+                    metadataValues.Add(LibraryIncludeFlagUtils.GetFlagString(privateAssetsFlag));
+                }
+                await _project.InstallPackageWithMetadataAsync(dependency.Name, dependency.LibraryRange.VersionRange, metadataElements, metadataValues);
+            }
+        }
+
+        private void MigrateRuntimes(PackageSpec packageSpec, Microsoft.Build.Evaluation.Project buildProject)
+        {
+            var runtimes = packageSpec.RuntimeGraph.Runtimes;
+            if (runtimes != null && runtimes.Count > 0)
+            {
+                var runtimeIdentifierValues = string.Join(";", runtimes.Keys);
+                buildProject.SetProperty("RuntimeIdentifiers", runtimeIdentifierValues);
+            }
+        }
+
+        private void RemoveProjectJsonReference(Microsoft.Build.Evaluation.Project buildProject)
+        {
+            var projectJsonItem = buildProject.GetItems("None")
+                .Where(t => string.Equals(t.EvaluatedInclude, Path.GetFileName(ProjectJsonFilePath), StringComparison.OrdinalIgnoreCase))
+                .FirstOrDefault();
+
+            if (projectJsonItem != null)
+            {
+                buildProject.RemoveItem(projectJsonItem);
+            }
+
+        }
+
+        private void CreateBackupAndSave(out string backupProjectFile, out string backupJsonFile)
```

Comment:
having all out parameters on this makes it seem like this is not the right pattern to use here